lib/modules/manager/sbt/extract.ts

```diff
@@ -289,6 +295,30 @@ const query = q.tree<Ctx>({
   postHandler: registryUrlHandler,
 });
 
+function extractUrl(line: string): string | undefined {
+  const urlMatch = line.match(sbtProxyUrlRegex);
+  return urlMatch?.[0];
+}
+
+export function extractProxyUrls(
+  content: string,
+  packageFile: string,
+): string[] {
+  const extractedProxyUrls: string[] = [];
+  logger.debug(`Parsing proxy repository file ${packageFile}`);
+  for (const line of content.split(newlineRegex)) {
+    if (line.match(regEx('https?'))) {
+      const url = extractUrl(line);
+      if (url && parseUrl(url)) {
+        extractedProxyUrls.push(url.trim());
+      }
```

Comment:
```suggestion
    const extraction = sbtProxyUrlRegex.exec(line)
    if (extraction?.groups?.proxy) {
        extractedProxyUrls.push(extraction.groups.proxy);
```
Use the regex directly to do this in one step.